File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int maxWeight = 0;
        int maxHeight = 0;
        for(int[] s: sizes){
            // 1.
            int weight = Math.min(s[0], s[1]);
            int height = Math.max(s[0], s[1]);
            
            // 2.
            maxWeight = Math.max(maxWeight, weight);
            maxHeight = Math.max(maxHeight, height);
        }
        
        int answer = maxWeight * maxHeight;
        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # width가 항상 height보다 길다고 가정
    max_width = 0
    max_height = 0
    
    for w,h in sizes:        
        max_width = max(max_width, max(w,h))
        max_height = max(max_height, min(w,h))
        
    return max_width * max_height